driver-core/src/main/java/com/datastax/driver/core/policies/TokenAwarePolicy.java

```diff
@@ -250,9 +249,14 @@ protected Host computeNext() {
       final Iterator<Host> replicasIterator;
 
       if (replicaOrdering == ReplicaOrdering.RANDOM) {
-        List<Host> replicasList = Lists.newArrayList(replicas);
-        Collections.shuffle(replicasList, ThreadLocalRandom.current());
-        replicasIterator = replicasList.iterator();
+        if (replicas instanceof ImmutableList) {
+          ArrayList<Host> replicasCopy = new ArrayList<>(replicas);
+          Collections.shuffle(replicasCopy, ThreadLocalRandom.current());
+          replicasIterator = replicasCopy.iterator();
+        } else {
+          Collections.shuffle(replicas, ThreadLocalRandom.current());
+          replicasIterator = replicas.iterator();
+        }
```

Comment:
So the tablet map returns ImmutableList in all cases and it seems TokenMap does too. Currently the else branch is not run for anything if i understand correctly.
Why not make `TabletMap#getReplicas` return mutable list? The returned list does not come from its internals and has to be constructed anyway. It is fine to shuffle it later on because it does not impact the tablet map. This would result in copying 1 list less for the queries using tablet map.

Reply:
Later we will change it to hold immutable lists, no reason to do same work twice.

Reply:
removed `else` branch.